core/src/main/scala/org/apache/spark/sql/SnappyParser.scala

```diff
@@ -663,7 +673,7 @@ class SnappyParser(session: SnappySession)
     SELECT ~ (DISTINCT ~> trueFn).? ~
     (namedExpression + commaSep) ~
     (FROM ~ ws ~ relations).? ~
-    (WHERE ~ expression).? ~
+    (WHERE ~ (test(setTokenize) ~ expression ~ test(setNoTokenize))).? ~
```

Comment:
@kneeraj I still do think that some of the literals in projections should be replaced. Can you tell the analyzer exception with that in more detail? Perhaps we will have to make it selective as mentioned by Soubhik.

@soubhik-c Constants in casts are digits and not literals. I think substr should be fine with replacement since it already accepts a generic expression (and ParamLiteral is just another type of it). Can you list down other cases you remember from gemxd work?

@hbhanawat and others: this was the primary reason I was not inclined to throw away our parser and switch to Spark's parser. However, it also means that it will be much more difficult to move to Spark's parser once this is merged. To do the move we will need to rewrite the tokenization perhaps with hand-crafted lexer/parser as was done for gemxd (and which caused lots of trouble since it had very simplistic state machine for semantics). Without that the performance of Spark's parser is not upto mark (in the range of ~0.2-0.4ms from what I remember as opposed to ~0.02-0.05ms for ours). We will have to compare that work and the additional DDL etc clauses vs keeping track of features being added to Spark's parser and add those to ours.